server-database/helpers.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"fmt"
+	"image"
+	_ "image/gif"
+	_ "image/jpeg"
+	_ "image/png"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/adrg/strutil"
+	"github.com/adrg/strutil/metrics"
+)
+
+func ValidateIndent(indent string) (int, error) {
+	i, err := strconv.Atoi(indent)
+	if err != nil {
+		log.Printf("Unable to parse indent: %s", err.Error())
+		return 0, fmt.Errorf("Unable to parse indent: %s", indent)
+	}
+	if i < 0 {
+		return 0, fmt.Errorf("Indent cannot be negative: %s", indent)
+	}
+	return i, nil
+}
+
+func ValidateAltText(title, altText string) error {
```

Comment:
Can you talk through your thinking behind what makes "good" alt text? What are you testing here? What would you _ideally_ test? What's easy to test? What's hard to test? What are the risks of going too far in either direction (either being too strict, or too lax)?

Reply:
So what I tried to do is to check for word similarity and allow if there's at least 1 word in the alt text that is _similar_ to a word in the title. 

I think it's very light checking which will maybe work 50% of the time.

What would we ideally test? Ideally we should test if title and alt text are related, if they can be used interchangeably. 

What's easy to test? Word equality is easy, word similarity is harder but it's still easy
What's hard to test? Relation between words or text, meaning of words or text. Text can be similar or even equal but mean completely different things. 

Being too strict could result in our solution rejecting valid option as it might not understand what humans do. A Solution that's too strict (possibly involving ML could also be computationally expensive). 
Being too lax could result in images having irrelevant alt texts

Reply:
> Ideally we should test if title and alt text are related, if they can be used interchangeably.

I'm not sure this is actually our ideal. While the title and alt-text probably are related, I can imagine them having very different forms.

I can imagine a title being quite brief and punchy: "The cutest cat" or "Cat makes a surprising friend", and the alt-text being much more verbose and descriptive: "A cat climbing up a tree hugging a smiling squirrel which has wrapped its tail around the cat".

These two fields are performing really different functions. There's not necessarily a correspondence between the word "cutest" or "surprising" in the title and any word in the alt-text - they have different purposes.

I'd say our ideal alt text is "an accurate and descriptive description of the contents of the image". _But_ this is an incredibly difficult problem to solve! (In fact, if it was easy to solve, we wouldn't need alt-text, a computer could just generate it for us).

Keying off of the title makes a lot of sense as an easy source of reference data - good thinking to look there! But because it's just a vague reference source, and not an oracle of good data, I'd lean in your implementation towards being very not-strict, and tending towards allowing maybe-bad alt-text because it's hard to test for.